src/Share.jsx

```diff
@@ -1,71 +1,74 @@
-import React, {createContext, useState, useMemo} from 'react'
-import {
-  Outlet,
-  Routes,
-  Route,
-  useLocation,
-  useNavigate,
-} from 'react-router-dom'
+import React, {
+  createContext,
+  useMemo,
+  useEffect,
+  useState,
+} from 'react'
+import {useNavigate, useParams} from 'react-router-dom'
 import {ThemeProvider, createTheme} from '@mui/material/styles'
 import CadView from './Containers/CadView'
 import debug from './utils/debug'
 import 'normalize.css'
 
+// TODO: This isn't used.
+// If icons-material isn't imported somewhere, mui dies
+/* eslint-disable */
+import AccountCircle from '@mui/icons-material/AccountCircle'
+/* eslint-enable */
+
 
 /**
- * Forward page from /share to /share/v/p per spect at:
- *   https://github.com/buildrs/Share/wiki/URL-Structure
- * @return {Object}
+ * Handles path demuxing to pass to CadView.
+ * @param {string} installPrefix e.g. '' on bldrs.ai or /Share on GitHub pages.
+ * @param {string} appPrefix e.g. /share is the prefix for this component.
+ * @param {string} pathPrefix e.g. v/p for CadView, currently the only child.
+ * @return {Object} The Share react component.
  */
-function Forward({appPrefix}) {
-  const location = useLocation()
+export default function Share({installPrefix, appPrefix, pathPrefix}) {
   const navigate = useNavigate()
-  React.useEffect(() => {
-    debug().log('Share.jsx: should forward?: ', location)
-    if (location.pathname == appPrefix) {
-      const dest = appPrefix + '/v/p'
-      debug().log('Share.jsx: Base: forwarding to: ', dest)
-      navigate(dest)
+  const urlParams = useParams()
+  const [modelPath, setModelPath] = useState(null)
+  const [mode, setMode] = useState('light')
+
+
+  /**
+   * On a change to urlParams, setting a new model path will clear the
+   * scene and load the new model IFC.  If there's not a valid IFC,
+   * the helper will redirect to the index file.
+   *
+   * Otherwise, the param change is a sub-path, e.g. the IFC element
+   * path, so no other useEffect is triggered.
+   */
+  useEffect(() => {
+    onChangeUrlParams()
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
It seems the array of dependencies has been defined, do you still need a disable rule?